lib/connectors_async/job_runner.rb

```diff
@@ -51,6 +62,30 @@ def log(str)
 
     private
 
+    def with_throttling(job)
+      attempts = 0
+      if job.should_wait?
+        log("Job #{job.id} is sleeping: Enterprise Search hasn't picked up documents for a while.")
+
+        while job.should_wait?
+          if attempts < ConnectorsShared::Constants::MAX_IDLE_ATTEMPTS
+            attempts += 1
+            idle(ConnectorsShared::Constants::IDLE_SLEEP_TIME)
+          else
+            raise JobStuckError.new("Enterprise Search failed to collect the data from the queue, waited #{attempts} times for #{IDLE_SLEEP_TIME} seconds.")
+          end
+        end
+
+        log("Job #{job.id} woke up")
+      end
+
+      yield
+    end
+
+    def idle(time)
```

Comment:
why the alias ? can't you just use `sleep` ?

Reply:
Oh, I just forgot a logger here :)

Reply:
mmm.. it still strikes me as overkill to create a whole method just to call `sleep`